CMakeLists.txt

```diff
@@ -23,11 +23,15 @@ endif()
 
 if(DEFINED ENV{INTELROOT})
   set(INTEL_ROOT_DEFAULT $ENV{INTELROOT})
+elseif(DEFINED ENV{ONEAPI_ROOT})
+  set(INTEL_ROOT_DEFAULT $ENV{ONEAPI_ROOT})
```

Comment:
Shouldn't the Intel root be set to `$ENV{ONEAPI_ROOT}/..`?

Reply:
I think this will keep the compatibility with older installations of MKL where the environment variable was `INTELROOT`.

Reply:
Sorry, seems I misread the issue. Checking now.

Reply:
The env points in `ProgramFiles(x86)\Intel\oneAPI` so it seems correct and consistent with the other path changes.

Reply:
I think it's confusing that `INTEL_ROOT` points to the oneAPI directory on Windows, but not on Mac and Linux.

That's why I would suggest to set `INTEL_ROOT` to the parent directory of `ONEAPI_ROOT` and explicitly include `oneAPI` in Windows search paths like it is done for other platforms, e.g.

```
${INTEL_ROOT}/oneAPI/compiler/latest/windows/compiler/lib/intel64_win
${INTEL_ROOT}/oneAPI/mkl/latest
```

Is this possible?

Reply:
Yes, that would be more consistent, so I made the changes. I also removed `oneAPI` from this path: 
```
$ENV{${ProgramFilesx86}}/Intel/oneAPI)
```
Still, in Mac the `$ONEAPI_ROOT` env also points in `/opt/intel/oneapi`. Isn't the same in Linux?

Reply:
> Still, in Mac the $ONEAPI_ROOT env also points in /opt/intel/oneapi. Isn't the same in Linux?

I think it's the same. But in the CMakeLists.txt we set `INTEL_ROOT`, not `ONEAPI_ROOT`. So we should set the following values:

* `INTEL_ROOT` -> `/opt/intel`
* `ONEAPI_ROOT` -> `/opt/intel/oneapi`
* `MKL_ROOT` -> `/opt/intel/oneapi/mkl/latest`

Your last change is looking good in that regard. Thanks!